main/src/main/scala/sbt/VirtualAxis.scala

```diff
@@ -66,11 +66,10 @@ object VirtualAxis {
     isScala2Scala3Sandwich(str(sbv1), str(sbv2))
   }
 
-  private[sbt] def isScala2Scala3Sandwich(sbv1: String, sbv2: String): Boolean = {
+  private[sbt] def isScala2Scala3Sandwich(sbv1: String, sbv2: String): Boolean =
     def compare(a: String, b: String): Boolean =
-      a == "2.13" && (b.startsWith("0.") || b.startsWith("3."))
+      a == "2.13" && (b == "3" || b.startsWith("3."))
```

Comment:
Main fix is this line.

Reply:
Can we just check `b.startsWith("3")` instead of `(b == "3" || b.startsWith("3."))`? Or would that mishandle some edge case?

Reply:
It will break when Scala 30.x comes out in the future.